### PR TITLE
Make sure actor context is a vailable for post stop

### LINKED ISSUE #26586: Sort of failed: context accessed before typed actor started
It doesn't fail any tests but it must be a consequence of some recent change (poststop?) that causes a few tests in the typed test suite to log this:
```
java.lang.IllegalStateException: Context was accessed before typed actor was started.
	at akka.actor.typed.internal.adapter.ActorAdapter.ctx(ActorAdapter.scala:80)
	at akka.actor.typed.internal.adapter.ActorAdapter.postStop(ActorAdapter.scala:253)
	at akka.actor.Actor.aroundPostStop(Actor.scala:558)
	at akka.actor.Actor.aroundPostStop$(Actor.scala:558)
	at akka.actor.typed.internal.adapter.ActorAdapter.aroundPostStop(ActorAdapter.scala:69)
	at akka.actor.dungeon.FaultHandling.finishTerminate(FaultHandling.scala:214)
	at akka.actor.dungeon.FaultHandling.terminate(FaultHandling.scala:172)
	at akka.actor.dungeon.FaultHandling.terminate$(FaultHandling.scala:142)
	at akka.actor.ActorCell.terminate(ActorCell.scala:445)
	at akka.actor.ActorCell.invokeAll$1(ActorCell.scala:551)
	at akka.actor.ActorCell.systemInvoke(ActorCell.scala:567)
	at akka.dispatch.Mailbox.processAllSystemMessages(Mailbox.scala:293)
	at akka.dispatch.Mailbox.run(Mailbox.scala:228)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:241)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #26586

### CODE REVIEW

File: akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala [hunk 1]
```diff
@@ -222,7 +222,9 @@ import akka.annotation.InternalApi
       case null                   => // skip PostStop
       case _: DeferredBehavior[_] =>
       // Do not undefer a DeferredBehavior as that may cause creation side-effects, which we do not want on termination.
-      case b => Behavior.interpretSignal(b, ctx, PostStop)
+      case b =>
+        if (_ctx eq null) initializeContext()
```
Comment (patriknw) [inline akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala]: could you repeat why it's important to not create the context in all cases from `preStart`?

Comment (patriknw) [reply akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala]: note that `GuardianActorAdapter` is also messing with `initializeContext()`, I think it needs a holistic cleanup

Comment (patriknw) [reply akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala]: and when cleaning up that also take a look at `protected var behavior` and see if that can be made `private`, now that we have `def currentBehavior`

Comment (johanandren) [reply akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala]: Yeah, I've been on the fence about doing a bigger rework of it, I don't like it at all how hard it is to follow, but I also remember that it was really tricky to get it behave like we want it to when it was created.

I wonder if we could _just_ make it a stashing behavior instead and not have a special adapter at all... (I have a feeling I tried that back then though)

Comment (johanandren) [reply akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala]: Re the avoiding of creating the context I have no idea, let's create it all the time.

**Comment by johanandren** [comment]
Superseded by #26590

**Pull request closed without merging**